source/plugins/go/input/perf/perf.go

```diff
@@ -153,6 +165,14 @@ func (p perfPlugin) enumerate() ([]map[string]interface{}, []map[string]interfac
 
 	FLBLogger.Printf("perf::enumerate : Begin processing @ %s", time.Now().UTC().Format(time.RFC3339))
 
+	defer func() {
+		if r := recover(); r != nil {
```

Comment:
when this will be invoked? what's goal of this code to get the call stack when there is panic? isnt that be captured as part of the console logs to traces in AI?

Reply:
goal is to fail gracefully if a panic happens in the input plugin and not crash fluent-bit completely. also want to use the exception paradigm instead of writing to traces as logs

also if panic happens and fluent-bit crashes there will be no one to send the trace data

Reply:
are you saying the with this pattern only specific plugin (which has the issue) crashed and all other plugins runs fine?

Reply:
yes this will allow other components of fb to keep running